pallets/external-agents/src/lib.rs

```diff
@@ -348,8 +356,15 @@ impl<T: Trait> Module<T> {
             match (*slot, group) {
                 // Identity transition. No change in count.
                 (Some(AgentGroup::Full), Some(AgentGroup::Full)) => {}
-                // Demotion / Removal. Count is decrementing.
-                (Some(AgentGroup::Full), _) => Self::dec_full_count(ticker)?,
+                // Demotion. Count is decrementing.
+                (Some(AgentGroup::Full), Some(_)) => Self::dec_full_count(ticker)?,
+                // Removal. Count is decrementing.
+                (Some(group), None) => {
+                    if let AgentGroup::Full = group {
+                        Self::dec_full_count(ticker)?;
+                    }
+                    AgentOf::remove(agent, ticker);
+                }
```

Comment:
This could just be `if group.is_none() { AgentOf::remove(agent, ticker); }` just before the `Ok(())` and then the `match` can be left untouched.